### ISSUE TITLE
Keep same random seed when restarting a run

### ISSUE THREAD

**Issue by anaprietonem** [ecmwf/anemoi-core#2]
### What happened?

Current AnemoiTrainer defines the initial seed according to [    def initial_seed(self) -> int:](https://github.com/ecmwf/anemoi-training/blob/38b75fadd5fcdf935547e8239180a9280158a12d/src/anemoi/training/train/train.py#L99). When resuming a run this seed is not reloaded as part of the mlflow hyperparameters and hence a new random seed is generated. For consistency and reproducibility of the results, we would need to keep the same seed.

### What are the steps to reproduce the bug?

Train a model in Leonardo and resume that after 24 hrs - see that the resumed job has a different seed. 

### Version

Current

### Platform (OS and architecture)

ATOS

**Comment by ssmmnn11** [comment]
What we would need to save is the state of the random generators, if we just re-seed with the same initial seed we would repeat the random numbers we have drawn until then. If we want to resume in the middle of an epoch then we would also need to save the state of the dataloader.
